### PR TITLE
add network filter that limits available sites

### PR DESCRIPTION
**Pull request by JackVCurtis**
This PR adds a Network filter with two options, PRESCIENT and ProNET. Each site is associated with one of these networks through a new constant, and when a network is selected the available sites for the site filter will be limited to the sites for that network.

Screenshot 2024-04-02 at 11 40 46 AM

### CODE REVIEW

**Comment by Elkrival** [comment]
@JackVCurtis  Can you try running this using a chart with data..? I pulled your branch and I get errors

Screenshot 2024-04-02 at 12 25 53 PM

Screenshot 2024-04-02 at 12 25 48 PM

**Comment by JackVCurtis** [comment]
Yeah, I'll try. My data set is weird and I've had a hard time getting data to come back, but thanks for the catch.

**Comment by JackVCurtis** [comment]
> @JackVCurtis Can you try running this using a chart with data..? I pulled your branch and I get errors
> 
> Screenshot 2024-04-02 at 12 25 53 PM Screenshot 2024-04-02 at 12 25 48 PM

I got a working chart for my data set and tracked down the bug: [93df04e](https://github.com/AMP-SCZ/dpdash/pull/722/commits/93df04e034b35d6a7967ceba1590fa0e2d829868)

Looks like the issue was a move from `==` to `===` without converting the value from string to number, introduced when we added linting/formatting.

File: views/forms/DropdownCheckboxGroup/index.jsx [hunk 1]
```diff
@@ -49,6 +55,17 @@ const DropdownCheckboxGroup = ({ label, initialValues, onChange }) => {
     setSelectedValue(options)
     onChange(label, options)
   }
+
+  useEffect(() => {
+    if (resetOptions) {
+      const selectedValues = Object.keys(initialValues).filter(
+        (k) => initialValues[k].value === 1
```
Comment (Elkrival) [inline views/forms/DropdownCheckboxGroup/index.jsx]: what is k..?

File: views/forms/ChartFilterForm/index.jsx [hunk 2]
```diff
@@ -23,8 +25,21 @@ const ChartFilterForm = ({ initialValues, onSubmit }) => {
       },
       {}
     )
+    if (label === 'networks') {
```
Comment (Elkrival) [inline views/forms/ChartFilterForm/index.jsx]: can we add constants for the string values like 'networks', 'sites', etc..

File: views/forms/ChartFilterForm/ChartFilterForm.css [hunk 3]
```diff
@@ -3,7 +3,7 @@
   width: 100%;
   gap: 10px;
   padding-bottom: 20px;
-  grid-template-columns: repeat(1fr, 4);
+  grid-template-columns: repeat(1fr, 5);
```
Comment (Elkrival) [inline views/forms/ChartFilterForm/ChartFilterForm.css:6]: I wonder if we should also add a maxWidth, the client expressed a dislike when the filters expand wider than the charts..

File: views/forms/DropdownCheckboxGroup/index.jsx (hunk 1 above)
Comment (JackVCurtis) [reply views/forms/DropdownCheckboxGroup/index.jsx]: key

File: views/forms/ChartFilterForm/index.jsx (hunk 2 above)
Comment (JackVCurtis) [reply views/forms/ChartFilterForm/index.jsx]: f3c6984

File: views/forms/DropdownCheckboxGroup/index.jsx (hunk 1 above)
Comment (JackVCurtis) [reply views/forms/DropdownCheckboxGroup/index.jsx]: 60c4c65

**Review by mikestone14 — APPROVED** [review]

**Pull request merged**